src/store/actions.js

```diff
@@ -281,7 +281,14 @@ export default {
 				)
 			}
 
-			await markMailboxRead(mailboxId)
+			const updated = Object.assign({}, mailbox)
+			updated.unread = 0
+
+			await markMailboxRead(mailboxId).then(() => {
```

Comment:
move the *then* code to the next line. you are awaiting the return of the async function already